pkg/stream/client_test.go

```diff
@@ -115,8 +115,19 @@ var _ = Describe("Streaming testEnvironment", func() {
 	It("Create two times Stream", func() {
 		Expect(testEnvironment.DeclareStream(testStreamName, nil)).NotTo(HaveOccurred())
 		err := testEnvironment.DeclareStream(testStreamName, nil)
-		Expect(err).To(HaveOccurred())
-		Expect(err).To(Equal(StreamAlreadyExists))
+		Expect(err).NotTo(HaveOccurred())
+		Expect(testEnvironment.DeleteStream(testStreamName)).NotTo(HaveOccurred())
+	})
+
+	It("Stream Exists", func() {
+		exists, err := testEnvironment.StreamExists(testStreamName)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(exists).To(Equal(false))
+
+		Expect(testEnvironment.DeclareStream(testStreamName, nil)).NotTo(HaveOccurred())
```

Comment:
How about declaring the stream again to ensure that it does not return an error?

Then maybe another test, changing the configs, to ensure that it returns the precondition error.

Reply:
there are already two tests for that:
```
Create two times Stream
```
and
```
Create two times Stream precondition fail
```

Reply:
Oh, right! When I looked at the above test using the cropped github view I didn't realize that the stream had been created twice on that test already. Sounds good :)